Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -3720,6 +3720,27 @@ LayoutUnit RenderBox::constrainBlockMarginInAvailableSpaceOrTrim(const RenderBox
     return marginSide == MarginTrimType::BlockStart ? minimumValueForLength(style().marginBeforeUsing(&containingBlock.style()), availableSpace) : minimumValueForLength(style().marginAfterUsing(&containingBlock.style()), availableSpace);
 }
 
+bool RenderBox::hasTrimmedMargin(MarginTrimType marginTrimType) const
+{
+    ASSERT(!needsLayout());
+    auto* container = containingBlock();
+
+    // FIXME: Remove these asserts as they are implemented for each layout context
+    if (isFlexItem() || isGridItem())
+        ASSERT_NOT_IMPLEMENTED_YET();
+    else if (container->isBlockContainer())
+        ASSERT(marginTrimType == MarginTrimType::BlockStart);
+    else
+        return false;
+
+    auto isAdjacentToContainingBlockEdge = [&]() {
+        return container->isBlockContainer() && container->logicalMarginBoxTopForChild(*this) == container->borderAndPaddingBefore(); 
+    };
+
+    return style().isDisplayBlockLevel() && !isFloating() 
+        && containingBlock()->style().marginTrim().contains(marginTrimType) && isAdjacentToContainingBlockEdge();
```

Comment:
containingBlock() -> container (or simply containingBlock, if you decide to change the variable name)

Reply:
shouldn't you check against inFlow instead? This seems to return true for statically positioned out-of-flow boxes.

Reply:
Yes good catch, changed.